lib/puppet/util/tcp_validator.rb

```diff
@@ -0,0 +1,36 @@
+require 'socket'
+require 'timeout'
+require 'ipaddr'
+
+module Puppet
+  module Util
+    class TcpValidator
```

Comment:
is there a reason you're putting this into the puppet::util namespace, rather than just creating our own?
it kinda feels… dirty…